include/eve/arch/x86/tags.hpp

```diff
@@ -42,8 +52,28 @@ namespace eve
   struct ssse3_   : sse3_   {};
   struct sse4_1_  : ssse3_  {};
   struct sse4_2_  : sse4_1_ {};
-  struct avx_     : sse4_2_ {};
-  struct avx2_    : avx_    {};
+
+  struct avx_     : sse4_2_
+  {
+    static constexpr std::size_t bits           = 256;
+    static constexpr std::size_t bytes          = 32;
+    static constexpr bool        is_bit_logical = detail::x86_logical_status;
+
+    template<typename Type>
+    static constexpr std::size_t expected_cardinal = bytes / sizeof(Type);
+  };
+
+  struct avx2_    : avx_ {};
+
+  struct avx512_  : avx2_
+  {
+    static constexpr std::size_t bits               = 512;
+    static constexpr std::size_t bytes              = 64;
+    static constexpr bool        is_bitwise_logical = detail::x86_logical_status;
```

Comment:
? I see one struct with `is_bit_logical` the other with `is_bitwise_logical`.
Not that either of them makes sense to me

Reply:
Obvious error is obvious.
The name is also maybe to rework. It's the indicator that the logical is either a regular register OR a compact bitmap register like mmask.

Reply:
`logical_use_regular_register` is maybe better

Reply:
This one structure was redundant